src/nxdoc/nuxeo-server/platform-services/comments.md

```diff
@@ -34,27 +35,74 @@ They hold:
 
 Annotations are stored as regular document of type `Annotation` which inherit from `Comment`. They hold in addition of comments:
+
 - blob xpath
 
 Like comments, annotations have the [`Annotation`](http://community.nuxeo.com/api/nuxeo/latest/javadoc/org/nuxeo/ecm/platform/comment/api/Annotation.html) interface to use them.
 
 Comments and annotations can have the facet `ExternalEntity`. When they have this facet, it can hold in addition:
+
 - external entity id
 - external entity
 - external entity origin</br>
-With this facet, it is possible to handle external content such as annotations made in [Nuxeo Enhanced Viewer]({{page space='nxdoc' page='nuxeo-enhanced-viewer'}}) for instance.
+With this facet, it is possible to handle external content such as annotations made in [ARender]({{page space='nxdoc' page='nuxeo-arender-connector'}}) for instance.
+
+### Current implementation - TreeCommentManager
+
+#### Storage
+
+Comments and annotations are stored as a tree structure in a folderish document of type `CommentRoot` which is located right under the commented document. Comments are _special children_ which means they are not copied with the commented document during copy operations but they are snapshot when the document is checked in.</br>
+Please note comments are not restored when a version of the commented document is checked out.
+
+#### Permissions
+
+Comments and annotations rely on the the [`ACL`](https://community.nuxeo.com/api/nuxeo/latest/javadoc/org/nuxeo/ecm/core/api/security/ACL.html) security model present in Nuxeo. People who have access to a commented document have read access to its comments and can edit their own comments and annotations.
+
+#### Events
+
+All events share the same event context. It contains:
+
+- the parent document in `parentComment` which may be a comment or the commented document
+- the comment in `comment_document`
+- the comment text in `comment`
+- the category `commentCategory` in `category`
+
+A `commentAdded` event is fired when creating a comment or annotation.
+
+A `commentUpdated` event is fired when updating a comment or annotation.
+
+A `commentRemoved` event is fired when deleting a comment or annotation.
+
+#### Autosubscription
+
+When a first comment is added to a document, the document author is subscribed automatically to the `commentAdded` event on the document.</br>
+When a user adds his first comment on a document. He is subscribed automatically to the `commentAdded` event on the document.</br>
+Autosubscription is turned on by default. It can be turned off by contributing like so:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<component name="org.nuxeo.ecm.platform.comment.service.notification.custom">
+  <extension target="org.nuxeo.runtime.ConfigurationService" point="configuration">
+    <property name="org.nuxeo.ecm.platform.comment.service.notification.autosubscribe">false<property>
+  </extension>
+</component>
+```
+
+### Legacy implementation - PropertyCommentManager
```

Comment:
I'm not sure we want to keep the legacy implementation on the master branch.

It is interesting to keep it on 10.10 as it is the default implementation and we also need to state about the needed migration for security purposes (on 10.10).

Reply:
I removed it, I'll put it on 10.10